staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -456,6 +457,62 @@ func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
 
+type fieldValidationDirective int
+
+const (
+	ignoreFieldValidation fieldValidationDirective = iota
+	strictFieldValidation
+	warnFieldValidation
+)
+
+// fieldValidation checks if the fieldValidation query parameter is set on the request,
+// and if so ensures that the parameter is valid and that the request has a valid
+// media type, because the list of media types that support field validation are a subset of
+// all supported media types (only json and yaml supports field validation).
+func fieldValidation(req *http.Request) (fieldValidationDirective, error) {
+	supportedMediaTypes := []string{"application/json", "application/yaml"}
+	supported := false
+	contentType := req.Header.Get("ContentType")
```

Comment:
Content-Type?

Reply:
Good catch, I've fixed and added tests for this